File: goal-parser-interpretation/goal-parser-interpretation.js
/**
 * @param {string} command
 * @return {string}
 */
var interpret = function(command) {
  let result = '';
  for (let i = 0; i < command.length; i++) {
    if (command[i] === '(' && command[i + 1] === 'a') {
      result += 'al';
      i += 3;
    } else if (command[i] === '(' && command[i + 1] === ')') {
      result += 'o';
      i += 1;
    } else {
      result += command[i];
    }
  }
  return result;
};

File: goal-parser-interpretation/goal-parser-interpretation.py
class Solution:
    def interpret(self, command: str) -> str:
        res = ''
        for i in range(len(command)):
          if command[i] == '(' and command[i + 1] == 'a':
            res += 'al'
          elif command[i] == '(':
            res += 'o' 
          elif command[i] == 'G':
            res += 'G'
        return res